R/grandStandardizedSolution.R

```diff
@@ -0,0 +1,239 @@
+#' Grand Standardized Solution
+#'
+#' Grand standardized solution of a two-stage path analysis model.
+#'
+#' @param model_list A list of string variable describing the structural path
+#'                   model, in \code{lavaan} syntax
+#' @param se A Boolean variable. If TRUE, standard errors for the grand
+#'                   standardized parameters will be computed.
+#' @param acov_par An asymptotic variance-covariance matrix for a fitted
+#'                 model object.
+#' @param free_list A list of model matrices that indicate the position of
+#'                  the free parameters in the parameter vector.
+#' @return A matrix of the standardized model parameters and standard errors.
+#' @export
+#'
+#' @example
+#' library(lavaan)
+#'
+#' ## A single-group, two-factor example
+#' mod1 <- '
+#'    # latent variables
+#'      ind60 =~ x1 + x2 + x3
+#'      dem60 =~ y1 + y2 + y3 + y4
+#'    # regressions
+#'      dem60 ~ ind60
+#' '
+#' fit1 <- sem(model = mod1,
+#'           data  = PoliticalDemocracy)
+#' grandStardardizedSolution(fit1)
+#'
+#' ## A single-group, three-factor example
+#' mod2 <- '
+#'     # latent variables
+#'       ind60 =~ x1 + x2 + x3
+#'       dem60 =~ y1 + y2 + y3 + y4
+#'       dem65 =~ y5 + y6 + y7 + y8
+#'     # regressions
+#'       dem60 ~ ind60
+#'       dem65 ~ ind60 + dem60
+#' '
+#' fit2 <- sem(model = mod2,
+#'             data  = PoliticalDemocracy)
+#' grandStardardizedSolution(fit2)
+#'
+#' ## A multigroup, two-factor example
+#' mod3 <- '
+#'   # latent variable definitions
+#'     visual =~ x1 + x2 + x3
+#'     speed =~ x7 + x8 + x9
+#'   # regressions
+#'     visual ~ c(b1, b1) * speed
+#' '
+#' fit3 <- sem(mod3, data = HolzingerSwineford1939,
+#'             group = "school",
+#'             group.equal = c("loadings", "intercepts"))
+#' grandStardardizedSolution(fit3)
+#'
+#' ## A multigroup, three-factor example
+#' mod4 <- '
+#'   # latent variable definitions
+#'     visual =~ x1 + x2 + x3
+#'     textual =~ x4 + x5 + x6
+#'     speed =~ x7 + x8 + x9
+#'
+#'   # regressions
+#'     visual ~ c(b1, b1) * textual + c(b2, b2) * speed
+#' '
+#' fit4 <- sem(mod4, data = HolzingerSwineford1939,
+#'             group = "school",
+#'             group.equal = c("loadings", "intercepts"))
+#' grandStardardizedSolution(fit4)
+
+
+grandStardardizedSolution <- function(fit, model_list = NULL,
+                                      se = TRUE, acov_par = NULL,
+                                      free_list = NULL, level = .95) {
+  if (is.null(model_list)) model_list <- lavTech(fit, what = "est")
```

Comment:
Need to either import the functions from `lavaan` or use `lavaan::lavTech()`. Please run CMD check to identify potential issues.